### PR TITLE
Replace go-bindata with //go:embed and upgrade to go 1.18

### PR DESCRIPTION
**Pull request by kinghrothgar**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**The PR fulfills these requirements:**
- [x] All tests are passing?
- [ ] New/updated tests are included?
- [x] If any static assets have been updated, has ui/bindata.go been regenerated?
- [x] Are there doc blocks for functions that I updated/created?

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
This is an attempt to revive this PR https://github.com/hound-search/hound/pull/392. As it as sat for a long time, there are new versions of go available. This PR does the following:
- Depends on this [PR](https://github.com/hound-search/hound/pull/429) as a newer version of alpine is needed that has go 1.18
- Updates go.mod, README, and CI to use go 1.18
- Drop unsupported go 1.14 from CI
- Switches from go-bindata to stdlib embed package
- Ran a `make node_modules` with the latest version of npm (8.13.2) which updated the `package-lock.json`

### CODE REVIEW

**Comment by kinghrothgar** [comment]
As far as [this discussion](https://github.com/hound-search/hound/pull/392#discussion_r605694518) of losing the fact that previously building this from main only required Golang and in the current state of this PR, you will now need npm also.

As far as I see it, we have three solutions. They're all independent of each other, so any number of them could be chosen:
- Convert dev / local build workflow to utilize Docker. I'm in strong support of this. The [Update package-lock.json with latest npm install](https://github.com/hound-search/hound/pull/430/commits/5f0410f744d51754f9beb7a4c15bf360464bca15) commit was required because I'm not sure what version of npm was needed or even how to get specific version of npm on OS X. This would make it easy to pin specific versions of golang / npm and would reduce the local requirements to do full deving from Docker, golang, npm, jest, to JUST Docker.
- Provide binaries with releases. I think this would just be nice anyways.
- Commit the ui build artifacts to the repo. This is the only one I'm kinda meh on but will defer to the maintainers.

**Comment by kinghrothgar** [comment]
As far as [this discussion](https://github.com/hound-search/hound/pull/392#discussion_r604381608), I believe I tested that this works. If you run two `make ui`'s in a row, the second one says there's nothing to do. If you then change anything in `ui/assets/**/*`, make correctly detected the need to do the rebuild.

**Review by dschott68 — APPROVED** [review]
This looks good to me.  Need to resolve merge conflicts.

**Pull request closed without merging**
